pkg/csi/service/wcpguest/controller.go

```diff
@@ -575,13 +575,28 @@ func controllerPublishForBlockVolume(ctx context.Context, req *csi.ControllerPub
 			if err == nil || time.Now().After(timeout) {
 				break
 			}
+			if err != nil {
+				log.Errorf("failed to update virtualmachine. Err: %v", err)
+			}
 			virtualMachine = &vmoperatortypes.VirtualMachine{}
 			if err := c.vmOperatorClient.Get(ctx, vmKey, virtualMachine); err != nil {
 				msg := fmt.Sprintf("failed to get VirtualMachines for the node: %q. Error: %+v", req.NodeId, err)
 				log.Error(msg)
 				return nil, csifault.CSIInternalFault, status.Errorf(codes.Internal, msg)
 			}
 			log.Debugf("Found virtualMachine instance for node: %q", req.NodeId)
+			for _, volume := range virtualMachine.Status.Volumes {
+				if volume.Name == req.VolumeId && volume.Attached && volume.DiskUuid != "" {
+					diskUUID = volume.DiskUuid
+					isVolumeAttached = true
+					log.Infof("Volume %q is already attached in the virtualMachine.Spec.Volumes. Disk UUID: %q",
+						volume.Name, volume.DiskUuid)
+					break
+				}
+			}
+			if isVolumeAttached {
+				break
+			}
 		}
 		if err != nil {
```

Comment:
This `err` will always be nil. The `err := ...` on 573 shadows the outer `err`, so if the Update fails and the timeout is reached, we'll proceed onward.